rclcpp/src/rclcpp/executors/static_executor_waitable.cpp

```diff
@@ -0,0 +1,264 @@
+    throw std::runtime_error("Received NULL memory strategy in executor waitable.");
+  }
+  memory_strategy_ = memory_strategy;
+  // Get this StaticExecutorWaitable shared pointer
+  this_shared_ptr_ = this_shared_ptr;
+
+  // Get memory strategy and executable list. Prepare wait_set_
+  execute();
+}
+
+void
+StaticExecutorWaitable::execute()
+{
+  // Fill memory strategy with entities comming from weak_nodes_
+  get_memory_strategy();
+  // Fill exec_list_ with entities comming from weak_nodes_ (same as memory strategy)
+  get_executable_list();
+  // Resize the wait_set_ based on memory_strategy handles (rcl_wait_set_resize)
+  prepare_wait_set();
+}
+
+void
+StaticExecutorWaitable::get_memory_strategy()
+{
+  memory_strategy_->clear_handles();
+  bool has_invalid_weak_nodes = memory_strategy_->collect_entities(weak_nodes_);
+
+  // Clean up any invalid nodes, if they were detected
+  if (has_invalid_weak_nodes) {
+    auto node_it = weak_nodes_.begin();
+    while (node_it != weak_nodes_.end()) {
+      if (node_it->expired()) {
+        node_it = weak_nodes_.erase(node_it);
+      } else {
+        ++node_it;
+      }
+    }
+  }
+
+  // Add the static executor waitable to the memory strategy
+  memory_strategy_->add_waitable_handle(this_shared_ptr_);
+}
+
+void
+StaticExecutorWaitable::get_executable_list()
+{
+  exec_list_.clear();
+
+  for (auto & weak_node : weak_nodes_) {
+    auto node = weak_node.lock();
+    if (!node) {
+      continue;
+    }
+    // Check in all the callback groups
+    for (auto & weak_group : node->get_callback_groups()) {
+      auto group = weak_group.lock();
+      if (!group || !group->can_be_taken_from().load()) {
+        continue;
+      }
+
+      group->find_timer_ptrs_if(
+        [this](const rclcpp::TimerBase::SharedPtr & timer) {
+          if(timer){
+            exec_list_.add_timer(timer);
+          }
+          return false;
+        });
+      group->find_subscription_ptrs_if(
+        [this](const rclcpp::SubscriptionBase::SharedPtr & subscription) {
+          if(subscription){
+            exec_list_.add_subscription(subscription);
+          }
+          return false;
+        });
+      group->find_service_ptrs_if(
+        [this](const rclcpp::ServiceBase::SharedPtr & service) {
+          if(service){
+            exec_list_.add_service(service);
+          }
+          return false;
+        });
+      group->find_client_ptrs_if(
+        [this](const rclcpp::ClientBase::SharedPtr & client) {
+          if(client){
+            exec_list_.add_client(client);
+          }
+          return false;
+        });
+        group->find_waitable_ptrs_if(
+          [this](const rclcpp::Waitable::SharedPtr & waitable) {
+          if(waitable){
+            exec_list_.add_waitable(waitable);
+          }
+            return false;
+        });
+    }
+  }
+
+  // Add the executor's waitable to the executable list
+  exec_list_.add_waitable(this_shared_ptr_);
```

Comment:
you can get rid of this line by making changes to the exec list.
keep in mind that the exec list is only used by static executor so you can change it as you want.
you can make sure that there is a special place for this object in the exec list, such that this is added only at the beginning and then it's never removed.